File: fibonacci/python/10_lucas/recursion.py
def recursion(v0, v1, n):
    if n < 0:
        for i in range(-1, n-1, -1):
            v0, v1 = (v1 - v0, v0)
        return v0
    if n == 0:
        return v0
    if n == 1:
        return v1
    for i in range(2, n+1):
        v1, v0 = (v0 + v1, v1)
    return v1

File: fibonacci/julia/10_lucas/recursion.jl
function recursion(v0, v1, n)
    if n < 0
        for _ in -1:-1:n
            (v0, v1) = (v1 - v0, v0)
        end
        v0
    elseif n == 0
        v0
    elseif n == 1
        v1
    else
        for _ in 2:n
            (v1, v0) = (v0 + v1, v1)
        end
        v1
    end
end
